zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -867,24 +867,24 @@ void readResponse(ByteBuffer incomingBuffer) throws IOException {
             ReplyHeader replyHdr = new ReplyHeader();
 
             replyHdr.deserialize(bbia, "header");
-            if (replyHdr.getXid() == PING_XID) {
-                LOG.debug(
-                    "Got ping response for session id: 0x{} after {}ms.",
+            switch (replyHdr.getXid()) {
+            case PING_XID:
+                LOG.debug("Got ping response for session id: 0x{} after {}ms.",
                     Long.toHexString(sessionId),
                     ((System.nanoTime() - lastPingSentNs) / 1000000));
                 return;
-            }
-            if (replyHdr.getXid() == AUTHPACKET_XID) {
+              case AUTHPACKET_XID:
+                LOG.debug("Got auth session id: 0x{}", Long.toHexString(sessionId));
                 if (replyHdr.getErr() == KeeperException.Code.AUTHFAILED.intValue()) {
```

Comment:
`LOG.debug` is before `return` at last, not just after `case`. It doesn't matter